frameworks/rooch-framework/sources/simple_rng.move

```diff
@@ -75,32 +75,70 @@ module rooch_framework::simple_rng {
         return value
     }
 
+    fun seed_with_count(count: u64): vector<u8> {
+        let base_seed = seed();
+        let count_bytes = bcs::to_bytes(&count);
+        let i = 0;
+        while (i < 8) {
+            let value = (*vector::borrow(&base_seed, i)) ^ (*vector::borrow(&count_bytes, i));
+            vector::insert(&mut base_seed, i, value);
+            i = i + 1;
+        };
```

Comment:
Use `vector::append`?

Reply:
I tested and found that bytes_to_u64 only takes the first 8 bytes of the output bytes, so appending directly has no effect.

Reply:
We should update the seed function and add the count to the data before hashing.

Reply:
Directly modifying the seed function won’t cause compatibility issues?

Reply:
The seed function is a private function; no compatibility issue

Reply:
OK

Reply:
@jolestar help review it again.